app/services/user_signup.rb

```diff
@@ -0,0 +1,52 @@
+class UserSignup
+  attr_reader :error_message
+
+  def self.perform(user:, invitation_token:, stripe_token:)
+    new(user, invitation_token, stripe_token).perform
+  end
+
+  def initialize(user, invitation_token, stripe_token)
```

Comment:
I would certainly recommend setting `invitation_token` to `nil` as default. Not all sign ups will be through invitation. Also its better when you build tests.